File: String/273IntegertoEnglishWords.cpp
class Solution {
public:
    vector<string> ones = {"", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten", "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen"};
    vector<string> tens = {"", "", "Twenty", "Thirty", "Forty", "Fifty", "Sixty", "Seventy", "Eighty", "Ninety"};
    string numberToWords(int num) {
        return num ? helper(num).substr(1) : "Zero";
    }
    string helper(int n){
        if(n>=1000000000) return helper(n/1000000000) + " Billion" + helper(n%1000000000);
        if(n>=1000000) return helper(n/1000000) + " Million" + helper(n%1000000);
        if(n>=1000) return helper(n/1000) + " Thousand" + helper(n%1000);
        if(n>=100) return helper(n/100) + " Hundred" + helper(n%100);
        if(n>=20) return " " + tens[n/10] + helper(n%10);
        if(n>=1) return " " + ones[n];
        return "";
    }
};

File: String/273integertoEnglishWords.py
class Solution(object):
    def numberToWords(self, num):
        """
        :type num: int
        :rtype: str
        """
        to19 = 'One Two Three Four Five Six Seven Eight Nine Ten Eleven Twelve ' \
               'Thirteen Fourteen Fifteen Sixteen Seventeen Eighteen Nineteen'.split()
        tens = 'Twenty Thirty Forty Fifty Sixty Seventy Eighty Ninety'.split()
        
        def check(word):
            if word < 20:
                return to19[word-1:word]
            if word < 100:
                return [tens[word/10-2]] + check(word%10)
            if word < 1000:
                return [to19[word/100-1]]+ ['Hundred'] + check(word%100)
            for p,w in enumerate(('Thousand','Million','Billion'),1):
                if word < 1000**(p+1):
                    return check(word/(1000**p)) + [w] + check(word%1000**p)
        return ' '.join(check(num)) or 'Zero'